tests/pubsub/testmcache.nim

```diff
@@ -5,19 +5,21 @@ import stew/byteutils
 import ../../libp2p/[peerid,
                      crypto/crypto,
                      protocols/pubsub/mcache,
-                     protocols/pubsub/rpc/message,
                      protocols/pubsub/rpc/messages]
+import ./utils
 
 var rng = newRng()
 
 proc randomPeerId(): PeerId =
   PeerId.init(PrivateKey.random(ECDSA, rng[]).get()).get()
 
+const MsgIdGenFail = "msg id gen failure"
+
 suite "MCache":
   test "put/get":
     var mCache = MCache.init(3, 5)
     var msg = Message(fromPeer: randomPeerId(), seqno: "12345".toBytes())
-    let msgId = defaultMsgIdProvider(msg)
+    let msgId = defaultMsgIdProvider(msg).expect(MsgIdGenFail)
```

Comment:
generally, `expect` expects a "positive" message: `.expect("it worked")` because it will negate it in the error string that it actually print - right now, the code reads as if we "expect message id generation failure" which is off

Reply:
Fixed in https://github.com/status-im/nim-libp2p/pull/697.